02-express-tutorial/04-express-app.js

```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const path = require('path');
+
+const app = express();
+
+//setup static and middleware
+app.use(express.static('./public'))
+
+app.get('/', (req, res) => {
+    res.sendFile(path.resolve(__dirname, './navbar-app/index.html'))
+})
+
+app.all('*', (req, res) => {
+    res.status(404).send('resource not found')
+})
+
+app.listen(5000, () => {
```

Comment:
Extra challenge: can you make it so that we can provide a custom port with an environment variable, and also change the log message accordingly?

In other words, change the port to 3000 if I run the program like this:

```bash
PORT=3000 node 04-express-app.js
```